libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClientFactory.kt

```diff
@@ -46,10 +47,9 @@ class RustMatrixClientFactory @Inject constructor(
     private val utdTracker: UtdTracker,
 ) {
     suspend fun create(sessionData: SessionData): RustMatrixClient = withContext(coroutineDispatchers.io) {
-        val client = getBaseClientBuilder(sessionData.sessionPath)
-            .homeserverUrl(sessionData.homeserverUrl)
+        val client = getBaseClientBuilder(sessionData.sessionPath, sessionData.passphrase)
+            .serverNameOrHomeserverUrl(sessionData.homeserverUrl)
```

Comment:
Note from this PR, but the FIXME below should be moved above the line 83 with `serverVersions`